### ISSUE TITLE
Random SIGSEGV issue when running Stryker Mutator

### ISSUE THREAD

**Issue by Chowarmaan** [stryker-mutator/stryker-js#3005]
**Summary**
I am unable to reliably reproduce this, but I did get a trace log when it happens.  As such, I do not have a sample I can send.  The issue occurs where the child process crashes unexpectedly.

`
(22023) ERROR Stryker Unexpected error occurred while running Stryker ChildProcessCrashedError: Child process [pid 22587] exited unexpectedly with exit code null (SIGSEGV). Stdout and stderr were empty.
    at ChildProcessProxy.handleUnexpectedExit (/home/sscott/Development/nestjs-emulator/node_modules/@stryker-mutator/core/dist/src/child-proxy/child-process-proxy.js:34:37)
    at ChildProcess.emit (node:events:378:20)
    at maybeClose (node:internal/child_process:1067:16)
    at Socket.<anonymous> (node:internal/child_process:453:11)
    at Socket.emit (node:events:378:20)
    at Pipe.<anonymous> (node:net:665:12) {
  innerError: undefined,
  pid: 22587,
  exitCode: null,
  signal: 'SIGSEGV'
}
`

**Stryker config**

[stryker.conf.js.txt](https://github.com/stryker-mutator/stryker-js/files/6810495/stryker.conf.js.txt)

**Test runner config**
[jest.config.js.txt](https://github.com/stryker-mutator/stryker-js/files/6810502/jest.config.js.txt)
[jest-base.config.js.txt](https://github.com/stryker-mutator/stryker-js/files/6810503/jest-base.config.js.txt)

**Stryker environment**

Global:

Local to Project:
```
├── @stryker-mutator/core": "~5.2.0",
├── @stryker-mutator/jest-runner@5.2.1
├── @stryker-mutator/typescript-checker@5.2.1
```

```
├── @types/jest@26.0.24
├── jest-express@1.12.0
├── jest-extended@0.11.5
├── jest@27.0.6
├── ts-jest@27.0.3
├── tsconfig-paths-jest@0.0.1
```

**Test runner environment**

npm test

which is:
jest --config ./jest.config.js --watchAll
stryker run --fileLogLevel trace

**Your Environment**

| software         | version(s)
| ---------------- | -------
| node             | 15.9.0
| npm              | 7.19.1
| Operating System | Ubuntu 21.04
| NestJS         | 8.0.2

**Add stryker.log**

[stryker.log](https://github.com/stryker-mutator/stryker-js/files/6810488/stryker.log)

**Comment by Chowarmaan** [comment]
Forgot to add TSConfig.json
[tsconfig.json.txt](https://github.com/stryker-mutator/stryker-js/files/6810516/tsconfig.json.txt)

**Comment by nicojs** [comment]
Hi @Chowarmaan thanks for the extensive bug report. I can see that it is related to typechecking in `@stryker-mutator/typescript-checker`. The root cause is difficult to find out, might be related to TypeScript internals, what version of TypeScript are you using?

As a workaround, you can run Stryker without the checker:

```diff
-	checkers: ['typescript'],
+	checkers: [],
```

To solve this (not the root cause, but the crash), we should implement a restart mechanism in the TypeChecker proxy class, like we already have in the TestRunner proxy class.

**Comment by Chowarmaan** [comment]
I am using Typescript 4.3.5, and it is installed globally if that matters.  I will remove the checker entry for now.  Thanks for the quick feedback.

**Comment by nicojs** [comment]
This feature just got released for Stryker 5.3 please give it a try
